sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -373,7 +373,7 @@ case class InsertIntoTable(
       // Note: The parser (visitPartitionSpec in AstBuilder) already turns
       // keys in partition to their lowercase forms.
       val staticPartCols = partition.filter(_._2.isDefined).keySet
-      Some(table.output.filterNot(a => staticPartCols.contains(a.name)))
+      Some(table.output.filterNot(a => staticPartCols.contains(a.name.toLowerCase)))
```

Comment:
visitPartitionSpec is used by 
```
ShowPartitionsCommand
LoadDataCommand
TruncateTableCommand
AlterTableAddPartitionCommand
AlterTableRenamePartitionCommand
AlterTableDropPartitionCommand
AlterTableSetLocationCommand
```

I peeked several class above, seems the case sensitivity is not handled well.